recipe-server/normandy/recipes/models.py

```diff
@@ -319,6 +346,39 @@ def save(self, *args, **kwargs):
         super().save(*args, **kwargs)
 
 
+class ApprovalRequest(models.Model):
+    revision = models.OneToOneField(RecipeRevision, related_name='approval_request')
+    created = models.DateTimeField(default=timezone.now)
+    creator = models.ForeignKey(User, on_delete=models.SET_NULL, related_name='approval_requests',
+                                null=True)
+    approved = models.NullBooleanField(null=True)
+    approver = models.ForeignKey(User, on_delete=models.SET_NULL, related_name='approved_requests',
+                                 null=True)
+
+    class AlreadyApproved(Exception):
```

Comment:
I think this name is a little misleading, because it could also be raised by a recipe that has been rejected (not approved). Maybe this should be renamed? `NotPending` might be good. Or maybe have both `AlreadyApproved` and `AlreadyRejected`?